airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java

```diff
@@ -53,15 +62,28 @@ public class BootloaderApp {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(BootloaderApp.class);
   private static final AirbyteVersion VERSION_BREAK = new AirbyteVersion("0.32.0-alpha");
+  private static final String DRIVER_CLASS_NAME = DatabaseDriver.POSTGRESQL.getDriverClassName();
 
   private final Configs configs;
+  private final DataSource configsDataSource;
+  private final DataSource jobsDataSource;
+  private final DSLContext configsDslContext;
+  private final DSLContext jobsDslContext;
   private Runnable postLoadExecution;
   private final FeatureFlags featureFlags;
 
   @VisibleForTesting
   public BootloaderApp(final Configs configs, final FeatureFlags featureFlags) {
     this.configs = configs;
     this.featureFlags = featureFlags;
+
+    // Hack to allow for existing tests to work with refactored code to support Dependency Injection
+    this.configsDataSource = DataSourceFactory.create(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(), DRIVER_CLASS_NAME,
+        configs.getConfigDatabaseUrl());
+    this.configsDslContext = DSLContextFactory.create(configsDataSource, SQLDialect.POSTGRES);
+    this.jobsDataSource =
+        DataSourceFactory.create(configs.getDatabaseUser(), configs.getDatabasePassword(), DRIVER_CLASS_NAME, configs.getDatabaseUrl());
+    this.jobsDslContext = DSLContextFactory.create(jobsDataSource, SQLDialect.POSTGRES);
   }
```

Comment:
I think this whole constructor implementation could be replaced with
```
@VisibleForTesting
  public BootloaderApp(final Configs configs, final FeatureFlags featureFlags) {
    this(configs, null, featureFlags);
  }
```
which should result in the same behavior while being DRY

Reply:
This has been changed.